harness/determined/common/experimental/trial.py

```diff
@@ -19,6 +31,61 @@ def __init__(self, trial_id: int, session: api.Session):
         self.id = trial_id
         self._session = session
 
+    def logs(
+        self,
+        follow: bool = False,
+        *,
+        head: Optional[int] = None,
+        tail: Optional[int] = None,
+        container_ids: Optional[List[str]] = None,
+        rank_ids: Optional[List[int]] = None,
+        stdtypes: Optional[List[str]] = None,
+        min_level: Optional[LogLevel] = None,
+    ) -> Iterable[str]:
+        """
+        Return an iterable of log lines from this trial meeting the specified criteria.
+
+        Arguments:
+            follow (bool, optional): If the iterable should block waiting for new logs to arrive.
+                Mutually exclusive with ``head`` and ``tail``.  Defaults to ``False``.
+            head (int, optional): When set, only fetches the first ``head`` lines.  Mutually
+                exclusive with ``follow`` and ``tail``.  Defaults to ``None``.
+            tail (int, optional): When set, only fetches the first ``head`` lines.  Mutually
+                exclusive with ``follow`` and ``head``.  Defaults to ``None``.
+            container_ids (List[str], optional): When set, only fetch logs from lines from
+                specific containers.  Defaults to ``None``.
+            rank_ids (List[int], optional): When set, only fetch logs from lines from
+                specific ranks.  Defaults to ``None``.
+            stdtypes (List[int], optional): When set, only fetch logs from lines from the given
+                stdio outputs.  Defaults to ``None`` (same as ``["stdout", "stderr"]``).
+            min_level: (LogLevel, optional): When set, defines the minimum log priority for lines
+                that will be returned.  Defaults to ``None`` (all logs returned).
+        """
+        if head is not None and head < 0:
+            raise ValueError(f"head must be non-negative, got {head}")
+        if tail is not None and tail < 0:
+            raise ValueError(f"tail must be non-negative, got {tail}")
+        for log in logs.trial_logs(
+            session=self._session,
+            trial_id=self.id,
+            head=head,
+            tail=tail,
+            follow=follow,
+            # TODO: Rename this to "node_id" and support it in the python sdk.
```

Comment:
How will you make sure this TODO is done?

Reply:
Honestly, my plan is not to bother.  This would require a big change in backend naming of things, and in our docs.  It'll probably be a long time before this is important enough to happen, long enough that any ticket I make is more likely to be forgotten than completed.

But it's useful to document why a particular option is _not_ being exposed in the public api.  In this case, it's named `agent_id` but secretly applies to all kinds of nodes other than agents.